src/main/scala/regmapper/RegMapper.scala

```diff
@@ -76,28 +76,28 @@ object RegMapper
 
     // Find the minimal mask that can decide the register map
     val mask = AddressDecoder(wordmap.keySet.toList)
-    val maskMatch = ~UInt(mask, width = inBits)
+    val maskMatch = ~mask.U(inBits.W)
     val maskFilter = toBits(mask)
     val maskBits = maskFilter.filter(x => x).size
 
     // Calculate size and indexes into the register map
     val regSize = 1 << maskBits
     def regIndexI(x: Int) = ofBits((maskFilter zip toBits(x)).filter(_._1).map(_._2))
-    def regIndexU(x: UInt) = if (maskBits == 0) UInt(0) else
+    def regIndexU(x: UInt) = if (maskBits == 0) 0.U else
       Cat((maskFilter zip x.asBools).filter(_._1).map(_._2).reverse)
 
-    val findex = front.bits.index & maskMatch
-    val bindex = back .bits.index & maskMatch
+    val findex = (front.bits.index & maskMatch.asUInt).asUInt
+    val bindex = (back .bits.index & maskMatch.asUInt).asUInt
```

Comment:
don't need `asUInt` here, if you wanna make IDEA happy, try this:
```suggestion
    val findex: UInt = (front.bits.index & maskMatch)
    val bindex: UInt = (back .bits.index & maskMatch)
```